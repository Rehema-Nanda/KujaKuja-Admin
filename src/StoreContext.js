// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
