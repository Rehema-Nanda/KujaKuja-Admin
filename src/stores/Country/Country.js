import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";

class Country {
    store = null;
    id = null;
    enabled = null;
    name = null;
    isoTwoLetterCode = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            enabled: this.enabled,
            name: this.name,
            iso_two_letter_code: this.isoTwoLetterCode,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
        };
    }

    updateFromJson(json) {
        this.enabled = json.enabled;
        this.name = json.name;
        this.isoTwoLetterCode = json.iso_two_letter_code;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteCountry(this.id);
        await this.store.removeCountry(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateCountry(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createCountry(json);
            this.id = responseData.id;
            await this.store.addCountry(this);
        }
    }
}
decorate(Country, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Country;
