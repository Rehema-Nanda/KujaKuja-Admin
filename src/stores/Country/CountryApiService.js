import ApiService from "../ApiService";

export default class CountryApiService extends ApiService {
    fetchAllCountries = async (config) => {
        const countriesResponse = await this.makeHttpCall("countries/paginator", "post", config);
        return countriesResponse.data; // TODO: change data structure returned by API
    };

    fetchEnabledCountries = async () => {
        const enabledCountriesResponse = await this.makeHttpCall("countries/enabled");
        return enabledCountriesResponse.data.data; // TODO: change data structure returned by API
    };

    fetchCountryById = async (id) => {
        const countryResponse = await this.makeHttpCall(`countries/${id}`);
        return countryResponse.data.data; // TODO: change data structure returned by API
    };

    deleteCountry = async (id) => {
        const deleteCountryResponse = await this.makeHttpCall(`countries/${id}`, "delete");
        // TODO: POST/PUT/DELETE endpoints should return string ID to be consistent with GET
        deleteCountryResponse.data.id = deleteCountryResponse.data.id.toString();
        return deleteCountryResponse.data;
    };

    createCountry = async (countryJson) => {
        const config = { data: countryJson };
        const createCountryResponse = await this.makeHttpCall("countries", "post", config);
        // TODO: POST/PUT/DELETE endpoints should return string ID to be consistent with GET
        createCountryResponse.data.id = createCountryResponse.data.id.toString();
        return createCountryResponse.data;
    };

    updateCountry = async (id, countryJson) => {
        const config = { data: countryJson };
        const updateCountryResponse = await this.makeHttpCall(`countries/${id}`, "put", config);
        // TODO: POST/PUT/DELETE endpoints should return string ID to be consistent with GET
        updateCountryResponse.data.id = updateCountryResponse.data.id.toString();
        return updateCountryResponse.data;
    };
}