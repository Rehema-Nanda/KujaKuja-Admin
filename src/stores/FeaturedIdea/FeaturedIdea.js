import { decorate, computed, comparer } from "mobx";

class FeaturedIdea {
    store = null;
    id = null;
    idea = null;
    createdAt = null;
    updatedAt = null;
    settlementId = null;
    settlementName = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            idea: this.idea,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
            settlement_id: this.settlementId,
        };
    }

    updateFromJson(json) {
        this.idea = json.idea;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;
        this.settlementId = json.settlement_id;
        this.settlementName = json.settlement_name;
    }

    async delete() {
        if (!this.id) {
            return;
        }

        await this.store.apiService.deleteFeaturedIdea(this.id);
        await this.store.removeFeaturedIdea(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateFeaturedIdea(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createFeaturedIdea(json);
            this.id = responseData.id;
            await this.store.addFeaturedIdea(this);
        }
    }
}

decorate(FeaturedIdea, {
    asJson: computed({ equals: comparer.shallow }),
});

export default FeaturedIdea;
