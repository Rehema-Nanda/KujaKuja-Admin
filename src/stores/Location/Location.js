import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";
import circle from "@turf/circle";
import _ from "lodash";

class Location {
    store = null;
    id = null;
    name = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;
    countryId = null;
    countryName = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            name: this.name,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
            country_id: this.countryId,
        };
    }

    updateFromJson(json) {
        this.name = json.name;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;
        this.countryId = json.country_id;
        this.countryName = json.country_name;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && typeof this.geojson === "string") {
            this.geojson = JSON.parse(this.geojson);
        }
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }

        // TODO: https://trello.com/c/9pBBU3hU/962-some-locations-are-missing-geometry
        // if no geometry, draw a 5km radius circle around coords
        if (this.geojson && _.isEmpty(this.geojson.geometry)) {
            const locationCoords = [this.lng, this.lat];
            const radius = 5;
            const options = { units: "kilometers" };
            this.geojson = circle(locationCoords, radius, options);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteLocation(this.id);
        await this.store.removeLocation(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateLocation(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createLocation(json);
            this.id = responseData.id;
            await this.store.addLocation(this);
        }
    }
}

decorate(Location, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Location;
