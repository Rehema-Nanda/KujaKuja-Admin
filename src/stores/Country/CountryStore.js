import {
    flow, decorate, observable, computed,
} from "mobx";
import Country from "./Country";
import CountryApiService from "./CountryApiService";
import AppConfig from "../../AppConfig";

class CountryStore {
    countries = [];
    count = 0;
    enabledCountries = [];
    isLoading = false;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new CountryApiService(AppConfig.API_URL);

        this.loadCountries();
        this.loadEnabledCountries();
    }

    get countriesSortedByName() {
        return this.countries.slice().sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadCountries = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchAllCountries(config);
            this.countries.replace(countriesJson.data.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.count = countriesJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadEnabledCountries = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchEnabledCountries();
            this.enabledCountries.replace(countriesJson.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load enabled countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateCountryFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const countriesJson = yield this.apiService.fetchCountryById(id);
        let country = this.countries.find((c) => {
            return c.id === id;
        });

        if (countriesJson.length === 0 && country) {
            // country no longer exists on the server, remove it locally
            yield this.removeCountry(country);
            country = null;
        }
        if (countriesJson.length > 0) {
            if (country) {
                country.updateFromJson(countriesJson[0]);
            }
            else {
                country = new Country(this, countriesJson[0]);
                this.countries.push(country);
            }
        }
        this.isLoading = false;
        return country;
    });

    removeCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.splice(this.countries.indexOf(country), 1);
    });

    addCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.push(country);
    });
}
decorate(CountryStore, {
    countries: observable,
    enabledCountries: observable,
    countriesSortedByName: computed,
    isLoading: observable,
});

export default CountryStore;
