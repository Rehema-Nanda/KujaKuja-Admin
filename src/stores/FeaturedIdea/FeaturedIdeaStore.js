import { flow, decorate, observable } from "mobx";
import FeaturedIdea from "./FeaturedIdea";
import FeaturedIdeaApiService from "./FeaturedIdeaApiService";
import AppConfig from "../../AppConfig";

class FeaturedIdeaStore {
    featuredIdeas = [];
    lastFeaturedIdea = [];
    isLoading = false;
    count = 0;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new FeaturedIdeaApiService(AppConfig.API_URL);

        this.loadFeaturedIdeas();
        this.loadLastFeaturedIdea();
    }

    loadFeaturedIdeas = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;

        try {
            const featuredIdeasJson = yield this.apiService.fetchAllFeaturedIdeas(config);

            this.featuredIdeas.replace(featuredIdeasJson.data.map((featuredIdeaJson) => {
                return new FeaturedIdea(
                    this,
                    featuredIdeaJson,
                );
            }));
            this.count = featuredIdeasJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`FeaturedIdeaStore failed to load featured ideas. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadLastFeaturedIdea = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const lastIdeaJson = yield this.apiService.fetchFeaturedIdeaLastResponse();
            this.lastFeaturedIdea.replace(lastIdeaJson.map((ideaJson) => {
                return new FeaturedIdea(
                    this,
                    ideaJson,
                );
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`DashBoardStore failed to load last fearured idea. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateFeaturedIdeaFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const featuredIdeasJson = yield this.apiService.fetchFeaturedIdeaById(id);
        let featuredIdea = this.featuredIdeas.find((fi) => {
            return fi.id === id;
        });

        if (featuredIdeasJson.length === 0 && featuredIdea) {
            // featured idea no longer exists on the server, remove it locally
            yield this.removeFeaturedIdea(featuredIdea);
            featuredIdea = null;
        }
        if (featuredIdeasJson.length > 0) {
            if (featuredIdea) {
                featuredIdea.updateFromJson(featuredIdeasJson[0]);
            }
            else {
                featuredIdea = new FeaturedIdea(this, featuredIdeasJson[0]);
                this.featuredIdeas.push(featuredIdea);
            }
        }
        this.isLoading = false;
        return featuredIdea;
    });

    removeFeaturedIdea = flow(function* (idea) { // eslint-disable-line func-names, require-yield
        this.featuredIdeas.splice(this.featuredIdeas.indexOf(idea), 1);
    });

    addFeaturedIdea = flow(function* (idea) { // eslint-disable-line func-names, require-yield
        this.featuredIdeas.push(idea);
    });
}

decorate(FeaturedIdeaStore, {
    featuredIdeas: observable,
    lastFeaturedIdea: observable,
    isLoading: observable,
});

export default FeaturedIdeaStore;
