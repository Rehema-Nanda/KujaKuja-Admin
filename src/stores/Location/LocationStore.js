import {
    flow, decorate, observable, computed,
} from "mobx";
import Location from "./Location";
import LocationApiService from "./LocationApiService";
import AppConfig from "../../AppConfig";

class LocationStore {
    locations = [];
    locationsSyndicated = [];
    isLoading = false;
    count = 0;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new LocationApiService(AppConfig.API_URL);

        this.loadLocations();
        this.loadSyndicatedLocations();
    }

    get locationsSortedByName() {
        return this.locations.sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadLocations = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const locationsJson = yield this.apiService.fetchAllLocations(config);
            this.locations.replace(locationsJson.data.map((locationJson) => {
                return new Location(
                    this,
                    locationJson,
                );
            }));
            this.count = locationsJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`LocationStore failed to load locations. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadSyndicatedLocations = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const locationsJson = yield this.apiService.fetchSyndicatedLocations();

            this.locationsSyndicated.replace(locationsJson.map((locationJson) => {
                return new Location(
                    this,
                    locationJson,
                );
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`LocationStore failed to load syndicated locations. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateLocationFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const locationsJson = yield this.apiService.fetchLocationById(id);
        let location = this.locations.find((loc) => {
            return loc.id === id;
        });

        if (locationsJson.length === 0 && location) {
            // location no longer exists on the server, remove it locally
            yield this.removeLocation(location);
            location = null;
        }
        if (locationsJson.length > 0) {
            if (location) {
                location.updateFromJson(locationsJson[0]);
            }
            else {
                location = new Location(this, locationsJson[0]);
                this.locations.push(location);
            }
        }
        this.isLoading = false;
        return location;
    });

    removeLocation = flow(function* (location) { // eslint-disable-line func-names, require-yield
        this.locations.splice(this.locations.indexOf(location), 1);
    });

    addLocation = flow(function* (location) { // eslint-disable-line func-names, require-yield
        this.locations.push(location);
    });
}
decorate(LocationStore, {
    locations: observable,
    locationsSyndicated: observable,
    locationsSortedByName: computed,
    isLoading: observable,
});

export default LocationStore;
