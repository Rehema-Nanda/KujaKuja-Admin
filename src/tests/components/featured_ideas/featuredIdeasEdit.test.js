import React from "react";
import { mount } from "enzyme";
import FeaturedIdeasEdit from "../../../components/featured_ideas/edit";
import { StoreProvider } from "../../../StoreContext";
import CountryStore from "../../../stores/Country/CountryStore";
import Country from "../../../stores/Country/Country";
import LocationStore from "../../../stores/Location/LocationStore";
import Location from "../../../stores/Location/Location";
import FeaturedIdeaStore from "../../../stores/FeaturedIdea/FeaturedIdeaStore";
import FeaturedIdea from "../../../stores/FeaturedIdea/FeaturedIdea";

jest.mock("../../../stores/Country/CountryStore");
jest.mock("../../../stores/Location/LocationStore");
jest.mock("../../../stores/FeaturedIdea/FeaturedIdeaStore");

CountryStore.mockImplementation(() => {
    return {
        countries: [
            new Country(
                null,
                {
                    id: "1",
                    enabled: false,
                    name: "Afghanistan",
                    iso_two_letter_code: "AF",
                    geojson: null,
                    lat: "0.00000000",
                    lng: "0.00000000",
                    created_at: null,
                    updated_at: null,
                },
            ),
        ],
    };
});

const locations = [
    new Location(
        null,
        {
            id: "1",
            name: "Test Location",
            geojson: null,
            lat: "0.00000000",
            lng: "0.00000000",
            created_at: null,
            updated_at: null,
            country_id: "1",
            country_name: "Afghanistan",
        },
    ),
    new Location(
        null,
        {
            id: "1000000000",
            name: "Test Billion Id Location",
            geojson: null,
            lat: "0.00000000",
            lng: "0.00000000",
            created_at: null,
            updated_at: null,
            country_id: "1",
            country_name: "Afghanistan",
        },
    ),
];
LocationStore.mockImplementation(() => {
    return {
        locations: locations,
        locationsSortedByName: locations,
    };
});

const featuredIdeas = [
    new FeaturedIdea(
        null,
        {
            id: "1",
            idea: "Test Featured Idea",
            created_at: null,
            updated_at: null,
            settlement_id: "1",
            settlement_name: "Test Location",
        },
    ),
];

const mockFeaturedIdeaDelete = jest.fn();
featuredIdeas[0].delete = mockFeaturedIdeaDelete;

const mockUpdateFeaturedIdeaFromServer = jest.fn(() => {
    return featuredIdeas[0];
});

FeaturedIdeaStore.mockImplementation(() => {
    return {
        featuredIdeas: featuredIdeas,
        updateFeaturedIdeaFromServer: mockUpdateFeaturedIdeaFromServer,
    };
});

describe("FeaturedIdeasEdit component", () => {
    const testProps = {
        history: {
            push: jest.fn(),
        },
        match: {
            params: {
                id: 1,
            },
        },
    };
    const wrapper = mount(<FeaturedIdeasEdit {...testProps} />, {
        wrappingComponent: StoreProvider,
    });

    it("should render correctly", () => {
        expect(mockUpdateFeaturedIdeaFromServer).toHaveBeenCalled();
        wrapper.update();
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should push history when cancelForm function is called", () => {
        const e = { preventDefault: jest.fn() };
        const cancelForm = jest.spyOn(wrapper.instance(), "cancelForm");
        cancelForm(e);
        expect(testProps.history.push).toHaveBeenCalledWith("/featured_ideas");
    });

    it("should change modal state when toggleDeleteConfirmationModal function is called", () => {
        const modalState = wrapper.instance().simpleState.isDeleteConfirmationModalOpen;
        const toggleDeleteConfirmationModal = jest.spyOn(wrapper.instance(), "toggleDeleteConfirmationModal");
        toggleDeleteConfirmationModal();
        expect(wrapper.instance().simpleState.isDeleteConfirmationModalOpen).toEqual(!modalState);
    });

    it("should call delete function on Location object and push history when delete function is called", async () => {
        const deleteFn = jest.spyOn(wrapper.instance(), "delete");
        await deleteFn();
        expect(mockFeaturedIdeaDelete).toHaveBeenCalled();
        expect(testProps.history.push).toHaveBeenCalledWith("/featured_ideas");
    });

    xit("should only show locations with ID less than 1 billion", () => {
        // data that is syndicated from other environments is currently kept separate by assigning each source
        // environment an ID offset that is some multiple of 1Bn - see syndication task handlers in API for more
        // this prevents users from attaching featured ideas to syndicated locations
        const optionLength = wrapper.find("option").length;
        expect(optionLength).toBe(2);
        expect(wrapper.find("option").first().text()).toBe("Please select");
        expect(wrapper.find("option").last().text()).toBe("Test Location");
    });
});
