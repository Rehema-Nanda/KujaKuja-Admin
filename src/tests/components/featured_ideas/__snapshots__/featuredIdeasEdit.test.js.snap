// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedIdeasEdit component should render correctly 1`] = `
"<FeaturedIdeasEdit history={{...}} match={{...}}>
  <div className=\\"page\\">
    <div className=\\"title-background-parent\\">
      <div className=\\"title-background-shape\\" />
    </div>
    <Banner pageTitle=\\"featured idea\\" elementAction=\\"update\\" buttonFunction={[Function]} buttonDisabled={false}>
      <Container tag=\\"div\\">
        <div className=\\"container\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-7 col-md-8\\">
                  <div className=\\"page-title\\">
                    <h1 className=\\"left\\">
                      Featured idea
                    </h1>
                    <p>
                      Manage your
                       
                      featured idea
                    </p>
                  </div>
                </div>
              </Col>
              <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-5 col-md-4\\">
                  <div className=\\"page-title\\" />
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Container>
    </Banner>
    <ConfirmationModal buttonFunction={[Function]} toggle={[Function]} isOpen={false}>
      <Modal isOpen={false} toggle={[Function]} autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} />
    </ConfirmationModal>
    <Container tag=\\"div\\">
      <div className=\\"container\\">
        <div className=\\"white-page-bg\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col lg={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-lg-8 offset-lg-2\\">
                  <div>
                    <AvForm onValidSubmit={[Function]} tag={[Function: Form]} model={{...}} validationEvent={{...}} method=\\"get\\" onSubmit={[Function: onSubmit]} onKeyDown={[Function: onKeyDown]} onInvalidSubmit={[Function: onInvalidSubmit]}>
                      <Form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\" tag=\\"form\\">
                        <form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\">
                          <Row tag=\\"div\\">
                            <div className=\\"row\\">
                              <Col tag=\\"div\\" widths={{...}}>
                                <div className=\\"col\\">
                                  <AvField type=\\"select\\" name=\\"settlement_id\\" label=\\"Location\\" required={true} value=\\"1\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"settlement_id\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"settlement_id\\" className=\\"\\">
                                              Location
                                            </label>
                                          </Label>
                                          <AvInput id=\\"settlement_id\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"select\\" name=\\"settlement_id\\" required={true} value=\\"1\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"select\\" name=\\"settlement_id\\" required={true} value=\\"1\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"settlement_id\\">
                                              <select size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"settlement_id\\" required={true} value=\\"1\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"settlement_id\\" className=\\"is-untouched is-pristine av-valid form-control\\">
                                                <option value=\\"\\">
                                                  Please select
                                                </option>
                                                <option value=\\"1\\">
                                                  Test Location
                                                </option>
                                                <option value=\\"1000000000\\">
                                                  Test Billion Id Location
                                                </option>
                                              </select>
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"textarea\\" name=\\"idea\\" label=\\"Idea\\" rows=\\"5\\" required={true} value=\\"Test Featured Idea\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"idea\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"idea\\" className=\\"\\">
                                              Idea
                                            </label>
                                          </Label>
                                          <AvInput id=\\"idea\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"textarea\\" name=\\"idea\\" rows=\\"5\\" required={true} value=\\"Test Featured Idea\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"textarea\\" name=\\"idea\\" rows=\\"5\\" required={true} value=\\"Test Featured Idea\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"idea\\">
                                              <textarea size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"idea\\" rows=\\"5\\" required={true} value=\\"Test Featured Idea\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"idea\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                            </div>
                          </Row>
                          <BtnRow buttonLabel=\\"Update\\" cancelForm={[Function]}>
                            <Row className=\\"btn-row\\" tag=\\"div\\">
                              <div className=\\"btn-row row\\">
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\" />
                                </Col>
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\">
                                    <Button color=\\"primary\\" className=\\"primary cancel\\" onClick={[Function]} tag=\\"button\\">
                                      <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary cancel btn btn-primary\\" aria-label={{...}}>
                                        Back
                                      </button>
                                    </Button>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </BtnRow>
                        </form>
                      </Form>
                    </AvForm>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </div>
    </Container>
  </div>
</FeaturedIdeasEdit>"
`;
